### PR TITLE
DR 1129 Move prototype to new repo

### PR DESCRIPTION
**Pull request by AITR1**
**Description**

Moved prototype to new repo, detailed below are the actions included

### Copied the folders from uec-cm:
 - scripts
 - src/frontend
 - .gitignore
 - .gitleaks.toml
 - .gitleaksignore
 - .editorconfig
 - .ecrc
- .flake8

### frontend
- Changed file naming to follow standard conventions of kebab case, typically for typescript
-   Copied components, header and footer, removing tests.
  -  Removed custom styling for header and used transactional header from NHS-components-react instead
- Removed build hash from footer due to security hotspot and as testers won’t be using the prototype to leverage the commit hash. Requires updates to:
    - src/frontend/increment-version.js
    - src/frontend/src/buildconfig.json
    - src/frontend/src/components/footer/interface.ts
    - src/frontend/src/components/footer/index.tsx
    - src/frontend/src/App.tsx
- Renamed prototype page from src/frontend/src/prototype/index.tsx to src/frontend/src/prototype/login.tsx 
- Renamed “src/frontend/src/prototype/admin/index.tsx” to be “src/frontend/src/prototype/admin/admin-landing-page”
- Remove “/prototype” from routes
- Remove axe enablement in src/frontend/src/index.tsx
- Changed `role=main` to `<main>` in src/frontend/src/App.tsx
- Update to be ‘Capacity Management UI prototype’ in name in src/frontend/public/manifest.json
- Update title to 'UEC Capacity Management UI Prototype' in src/frontend/public/index.html
- Divided mock-data-service to 'src/frontend/src/mock-data/mock-hospital-data.ts', 'src/frontend/src/mock-data/mock-questionnaire-data.ts', 'src/frontend/src/mock-data/mock-user-data.ts'
- Changed names and emails in 'src/frontend/src/mock-data/mock-user-data.ts'
- Removed 'global.css' (makes version to left but no impact)
- Remove commented out code from src/frontend/src/pages/questionnaire-summary.tsx

 ### src/frontend/package.json
  - Updated name
  - Removed dependencies [yup, web-vitals, @testing-library/jest-dom, @testing-library/react, @testing-library user-event, @hookform/resolvers, ts-jest,  @axe-core/react, jest-fetch-mock, react-hook-form]
  - Moved type dependencies to devdependencies
    -@types/jest, @types/node, @types/react, @types/react-dom, @types/react-router-dom, typescript, react-scripts
    - Updated dependencies : 
        "@types/node": "^16.18.68", ->     "@types/node": "^22.7.4",
        "@types/react": "^18.2.45", ->  "@types/react": "^18.3.10",
        "@types/react-dom": "^18.2.18", -> "@types/react-dom": "^18.3.0" ***
    - Removed jest configuration
    - Removed scripts [test, test:ci,  test:coverage]

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,83 @@
...
+- Setup git locally on your machine and configure the git user email and name.
+- Setup gpg commit signing, which is required for commits to be 'verified'. Guidance on confluence
+- Run the pre-commit hooks located in the 'scripts' folder
+- Install an appropriate IDE compatible with Typescript
+- Install Node and NPM
+- Install and setup Docker
+- Clone the repository
+
+## Running the project
+
```
Comment (Tomdango) [inline README.md:35]: Is there a particular required Node.js version?

File: src/frontend/src/components/header/index.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
```
Comment (Tomdango) [inline src/frontend/src/components/header/index.tsx:1]: Just a thought - would it be simpler to have these as simply `src/components/header.tsx`?

File: src/frontend/src/css/prototype.css
Comment (Tomdango) [inline src/frontend/src/css/prototype.css:1]: To refactor as part of https://nhsd-jira.digital.nhs.uk/browse/DR-1110

File: src/frontend/src/pages/hospital-questionnaire.tsx
Comment (Tomdango) [inline src/frontend/src/pages/hospital-questionnaire.tsx:1]: (nothing to change for the prototype)
This page is shouting out that we need reusable validation and form logic in the proper build. We need to make sure we separate the business logic from the display logic - coupling these together results in these 800+ line files for a single page

**Review by Tomdango — APPROVED** [review]
Just a couple of nitpicks and observations. Looks good to go for the prototype - I also assume some of this might need tweaking regardless as part of DR-1130 but will be covered under that ticket.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,83 @@
...
+## Installation
+
+- Setup git locally on your machine and configure the git user email and name.
+- Setup gpg commit signing, which is required for commits to be 'verified'. Guidance on confluence
+- Run the pre-commit hooks located in the 'scripts' folder
+- Install an appropriate IDE compatible with Typescript
+- Install Node and NPM
+- Install and setup Docker
+- Clone the repository
+
```
[the remark this answers is not in the document]
Comment (AITR1) [reply README.md:34]: Updated

File: src/frontend/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
```
[the remark this answers is not in the document]
Comment (AITR1) [reply src/frontend/.eslintrc.json:1]: Updated

File: src/frontend/src/components/footer/interface.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export interface FooterProps {
```
[the remark this answers is not in the document]
Comment (AITR1) [reply src/frontend/src/components/footer/interface.ts]: Updated

File: src/frontend/src/components/header/index.tsx (hunk 2 above)
Comment (AITR1) [reply src/frontend/src/components/header/index.tsx]: Updated

File: src/frontend/src/buildconfig.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "buildDate": "22nd Mar 2024 14:17"
+}
```
[the remark this answers is not in the document]
Comment (AITR1) [reply src/frontend/src/buildconfig.json]: It is slightly misleading as the file doesn't get automatically updated during local development. Instead uses this date locally unless the prebuild step is run. 

So slight issue with removing this, the file is needed for src/frontend/increment-version.js. I've removed the   "buildDate" property from the file and have added the prebuild step of running the increment-version.js before the start package to update the builddate i.e "start": "npm run prebuild && react-scripts start". So the file is there but its empty and automatically added on the build start when developing locally.

However, adding it to the .gitignore after the change still picks up the changes to the file as its been committed. A couple of solutions, we could not commit the 'buildconfig.json' and have it as a readme step to create that file. Update the increment-version to create the file if not there. Or there is a command that can be used to not track a committed file, however it seems each dev would need to execute it (I think, not fully investigated) https://stackoverflow.com/questions/9794931/keep-file-in-a-git-repo-but-dont-track-changes.

File: src/frontend/src/mock-data/mock-questionnaire-data.ts [hunk 7]
```diff
@@ -0,0 +1,556 @@
...
+    ecmoPatients: 10,
+    invasivelyVentilatedPatients: 10,
+    nonInvasivelyVentilatedPatients: 5,
+    bedsOccupiedUnder1Year: 2,
+    bedsOccupiedBy12To17: 10,
+    bedsOccupiedBy18Plus: 5,
+    dischargesOrDeathExpectedNumberIn12Hours: 0,
+    yesterdaySurgeryCancellations: 5,
+    yesterdayRefusedUnplannedAdmissions: 5,
+    notDischargedPatientsForNonClenicalReasons: 2,
```
[the remark this answers is not in the document]
Comment (AITR1) [reply src/frontend/src/mock-data/mock-questionnaire-data.ts]: Updated

File: README.md (hunk 1 above)
Comment (AITR1) [reply README.md:38]: Updated to state lts node versin on readme

**Review by Tomdango — DISMISSED** [review]

File: src/frontend/src/buildconfig.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (Tomdango) [reply src/frontend/src/buildconfig.json]: You can try a `git rm src/frontend/src/buildconfig.json` - it won't automatically get picked up as a change as it's now in the .gitignore. Happy to resolve this if that doesn't work

[the remark this answers is not in the document]
Comment (AITR1) [reply src/frontend/src/buildconfig.json]: Understand - have reverted the changes I made so its acting like in the cm-repo again. Will resolve comment

**Pull request merged**